homeassistant/components/nest/config_flow.py

```diff
@@ -133,7 +141,14 @@ async def async_step_reauth_confirm(self, user_input=None):
     async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         if self.is_sdm_api():
-            if self.hass.config_entries.async_entries(DOMAIN):
+            # Only one config entry is allowed since only a single static
+            # config is currently allowed in configuration.yaml.  The config
+            # entry is allowed to be updated during reauth, otherwise there
+            # should be no existing configurations.
+            existing_entries = self.hass.config_entries.async_entries(DOMAIN)
+
+            assert len(existing_entries) <= 1, "Expected at most 1 config entry"
```

Comment:
I don't think we should add more assertions in the production code for all possible developer mistake scenarios. We're going to test all flow cases. That's enough.

Reply:
I agree with that spirit.  My impression is that this is not just a developer mistake, but that existing end users could have already added the integration twice before the check for duplicate config entries was added in this PR (even without the introduction of reauth).  That would be silly given they'd share the same config entry, so it is unlikely. 

I'm ok if the answer is: we still don't care, or something is already preventing this from happening, just wanted to clarify.

Reply:
If we want to handle this case we need to add clean up code that removes the config entries and let's the user configure one again. We can't rely on assert in production code.

Reply:
Thanks for talking through this.  I've added logic to cleanup while updating the existing entries silently which seems like it is in the spirit of what you're talking about.

Reply:
Would it be more suitable to do some cleanup using async_migrate_entry? I could imagine assigning a unique and deleting entries that are dupes, so simplify the config flow itself.

Reply:
`async_migrate_entry` should only be used when we need to change the entry data, it can be done automatically and it's not possible to avoid backwards incompatible changes of the entry data.

Most entry changes eg adding entry unique id can be done without migration allowing users to roll back if needed.